src/main.rs

```diff
@@ -27,35 +28,36 @@ run()
 
 fn main() {
     // init device
-    let device_handle = devices::tactrix_openport::new();
+    let device_handle = Arc::new(devices::tactrix_openport::new());
```

Comment:
we need to "share" the device handle between all incoming streams. So, we can't move its ownership into any individual thread. We put it inside an Arc (atomically reference-counted) smart pointer, which allocates some space for its contents on the heap. when you call `.clone` on an `Arc<T>`, you get another `Arc<T>` that points to the same `T` on the heap. When there are no `Arc<T>`s left for a given `T`, the `T` gets deallocated.